### DISCUSSION TITLE
Keyguard 2024 Recap 🎉

### DISCUSSION THREAD

**Discussion by AChep** [AChep/keyguard-app#755] [Announcements]
Hi everyone,

Here's a short overview of some of the changes that came to Keyguard in 2024. 

#### URL Overrides

With URL overrides you can write rules to create new URLs basing on the existing entries. The simplest and default example of it would be the HTTPS-ify URL override that automatically proposes you the same HTTP URL with schema changed to HTTPS.
 
<img width="471" alt="зображення" src="https://github.com/user-attachments/assets/dc0a2220-20dd-41b0-8d2f-22559dd2876f" />

You could also use this feature to create app specific command to launch something. For example, you can map SSH URI to a command that launches your SSH client.

[View docs](https://github.com/AChep/keyguard-app/blob/27b98f8d9d1bf70c89843c4f88fd4f331b7187ff/wiki/URL_OVERRIDE.md)

#### URL Placeholders

With URL placeholders you can dynamically use other fields from the app/item in the URL field: 
```
ssh://{username}@server.com
```

<img width="478" alt="зображення" src="https://github.com/user-attachments/assets/e185c881-98a7-42a4-8ff7-750bad5a6146" />

[View docs](https://github.com/AChep/keyguard-app/blob/27b98f8d9d1bf70c89843c4f88fd4f331b7187ff/wiki/PLACEHOLDERS.md)

### Custom Wordlists

With custom Wordlists you can upload your own dictionaries to later generate passphrases or usernames from. _Note that Keyguard will not display the correct passphrase strength when you use a custom wordlist._ 

[View docs](https://github.com/AChep/keyguard-app/blob/27b98f8d9d1bf70c89843c4f88fd4f331b7187ff/wiki/WORDLISTS.md)

### Export

You can export your vault (or the selected items) in a form of an encrypted ZIP archive, including file attachments! 

### Watchtower Alerts

Keyguard now keeps track of all previously seen alerts and notifies you if something new pops up! Much easier to keep track of it now!

<img width="637" alt="зображення" src="https://github.com/user-attachments/assets/cfc5026c-627b-412f-b994-06841dfb1437" />
<img width="213" alt="зображення" src="https://github.com/user-attachments/assets/1e3788e7-693a-4664-8733-04d4b7674081" />
  
### Custom filters

With custom filters you can now save your current filter state to a preset that you can later toggle. The custom filters pop up on top of the vault list for quick access.

### Sends

In 2024 Keyguard gained support for creating text sends!

### SSH keys

You can now view SSH items and generate new keys! The option is now available in the usual generator tab. 

<img width="461" alt="зображення" src="https://github.com/user-attachments/assets/96d7f4bb-1021-435e-980e-188f9095ec80" />

### Material Design refresh

We kept up with the latest changes in Material design, making use of surfaces in Keyguard.

<img width="1071" alt="зображення" src="https://github.com/user-attachments/assets/8700d2f5-5cb1-4bcc-8f3f-bcd448a37a44" />

### Equivalent domains

Keyguard now supports equivalent domains that you can configure in web vault. The equivalent domains list is applied to both autofill process as well as to all watchtower alerts.

### Package managers

Keyguard can now be found at [Homebrew](https://formulae.brew.sh/cask/keyguard), [scoop](https://github.com/ScoopInstaller/Extras) and [AUR](https://aur.archlinux.org/packages/keyguard).

### & more!

We introduced a lot of visual improvements, such as displaying OTP codes, attachments and passkeys directly in the vault list when you apply the appropriate filters, had a lot of quality of life updates for desktop and a ton of small features, such as an ability to scan Google Authenticator QR code for easier OTP migration!

Thanks for having me and happy holidays! 🎉

**Comment by JordyEGNL** [comment]
Thank you for your awesome work! Keyguard is now my main app for Bitwarden on my Phone 😄

**Comment by issamemohtadi** [comment]
Thank you . I hope to add attachments or photos from the keyguard application itself .

**Reply by AChep** [reply]
Yeah, seems like attachments are becoming the only large missing feature. I will definitely tackle it in near future.

**Comment by RubberDuckShobe** [comment]
Awesome work, this is probably one of the most polished apps ever!

**Comment by Vickstoen** [comment]
nice

**Comment by ThemanRonin** [comment]
Compared to the official Bitwarden client, Keyguard quality is vastly superior in every aspect. Thank you for developing such an excellent piece of software!

**Comment by Alex-wangwang** [comment]
good job
